### ISSUE TITLE
menu.popup() crashes the app in Electron 7.1.2 on Mac after opening the menu repeatedly

### ISSUE THREAD

**Issue by dgmid** [electron/electron#21338]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 7.1.2
* **Operating System:**
  * macOS 10.15.1
* **Last Known Working Electron version:**
  * 6.1.5

### Expected Behavior
Menu.popup() should always open without crashing the app

### Actual Behavior
repeated opening of the popup menu crashes the app

### To Reproduce

[https://github.com/dgmid/popup-test](https://github.com/dgmid/popup-test)

```sh
$ git clone https://github.com/dgmid/popup-test.git
$ npm install
$ npm start
```

### Additional Information
This problem only occurs with Electron v7x. Running with v6x or v8-beta works with no problems

**Comment by MarshallOfSound** [comment]
Fixed in #21225

**Comment by zcbenz** [comment]
@MarshallOfSound This crash happened with the fix, it seems to be a macOS Catalina only problem.

**Comment by zcbenz** [comment]
https://github.com/electron/electron/pull/20808#issuecomment-560237801

> [crash-report.txt](https://github.com/electron/electron/files/3909584/crash-report.txt)
> 
> I have also tested this on my old Macbook Pro which is running macOS 10.13.6 (High Sierra) and it doesn't crash that machine. Only the 10.15.1 (Catalina) machine.

**Comment by miniak** [comment]
@zcbenz v7.1.2 does not yet contain the fix, it's only going to be in v7.1.3
https://github.com/electron/electron/compare/v7.1.2...7-1-x
